test/integration/build_to_taskruns_test.go

```diff
@@ -202,4 +204,43 @@ var _ = Describe("Integration tests Build and TaskRun", func() {
 			})
 		})
 	})
+
+	Context("when a build with nodeSelector is defined", func() {
+		BeforeEach(func() {
+			buildSample = []byte(test.MinimalBuildWithNodeSelector)
+			buildRunSample = []byte(test.MinimalBuildRun)
+		})
+
+		Context("when the TaskRun is created", func() {
+			It("should have the nodeSelector specified in the PodTemplate", func() {
+				Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+				buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+				Expect(err).To(BeNil())
```

Comment:
Please add some `Expect` on the build status. This should help to identify why it does not validate as expected.

Reply:
Added some Expects here. After some research, I think for nodeSelectors kubernetes expects the keys to be qualified names, so I changed the validator to use:
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L42

Reply:
Reference: https://github.com/kubernetes/kubernetes/blob/v1.31.0/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go#L102